cli_reference/openshift_developer_cli/odo-disconnected-env.adoc

```diff
@@ -0,0 +1,480 @@
+[id='odo-disconnected-cluster']
+= Running odo in cluster disconnected from the Internet
+include::modules/openshift-developer-cli-attributes.adoc[]
+include::modules/common-attributes.adoc[]
+:context: odo-running-in-disconnected-envs
+toc::[]
+
+// TODO -> general steps for both java/nodejs
+// Question: Is only node.js and java supported for this?
+// TODO: windows
+// todo: do you have to be logged in to https://catalog.redhat.com/ to access to containers? (registry.redhat.io -> seems=yes, registry.access.redhat.com -> seems that login is not required). All links are registry.access.redhat.com so no login required.
+
+.Prerequisites
+* An {product-title} cluster in a restricted network
+* `oc` installed on the client OS
+// link?
+* `odo` installed on the client OS
+// link?
+* A mirror registry created for the disconnected cluster on the cluster's host OS
+* Certificate Authority (CA) certificate from the OpenShift cluster's host OS (`root` user)
+//link?
+// https://docs.openshift.com/container-platform/4.2/installing/install_config/installing-restricted-networks-preparations.html
+// TODO -> find docs how to setup a disconnected cluster
+// ^ yes they are there
+
+== Mirroring the `odo-init-image` into the cluster's local registry from the Red Hat registry
+//on the machine with cluster
+
+On the cluster's host OS:
+. Log in to the mirror registry with Docker.
++
+----
+$ docker login <mirror-registry>:<port> -u <mirror_registry_username> -p <mirror_registry_password>
+----
+// host has to be conected to the internet
+// not an issue if "setting up the cluster for usage in restricted network" requires an internet connection as well
+. Mirror the `odo-init-image` image into the local mirror registry.
++
+----
+$ oc image mirror registry.access.redhat.com/openshiftdo/odo-init-image-rhel7:latest <mirror-registry>:<port>/openshiftdo/odo-init-image-rhel7:latest
+----
+
+// can you connect the (openshift) host machines to the internet? (not the cluster itself)
```

Comment:
nope, that defeats the whole purpose of the disconnected cluster.

Reply:
@amitkrout so we have to devise a workaround -> ssh from odo machine maybe?